source/win-main/store.ts

```diff
@@ -265,12 +270,24 @@ function getConfig (): StoreOptions<ZettlrState> {
     },
     mutations: {
       updateTableOfContents: function (state, toc) {
+        // This function is called every time a document is changed
+        // (e.g., a letter is typed). Most of the time the headings
+        // stay the same. To avoid unnecessary computation, their
+        // parsed texts are cached:
+        // state.headingCache[unparsedHeadingText] = parsedHeadingText
+        const headingTextToHtml = function (headingText: string): string {
+          if (!state.headingCache[headingText]) {
```

Comment:
Here you can then use `if (!state.headingCache.has(headingText))`

Also, I'd suggest to refactor the for loop down below to something like this:

```js
for (const entry of toc) {
  if (!headingCache.has(entry.text) {
	headingCache.add(md2html(entry.text))
  }
  entry.text = headingCache.get(entry.text) as string
}
```

Makes it more concise